qlib/kernel/arch/aarch64/arch_def.rs

```diff
@@ -195,3 +195,67 @@ impl State {
         }
     }
 }
+
+pub type Context = Aarch64Context;
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct Aarch64Context {
```

Comment:
is this the context to use for all situations ?

Reply:
I'm not quite sure, @chl337 is taking care of the context swap thing, he can add or delete registers from this struct if it is necessary.

Reply:
As I understand it, this structure is used for task switching, e.g. after a scheduling decision... That said, AFAIK at least the _calle-saved, FP, LR, SP_ registers must be preserved, **ref:** [ABI](https://github.com/ARM-software/abi-aa/blob/2a70c42d62e9c3eb5887fa50b71257f20daca6f9/aapcs64/aapcs64.rst#general-purpose-registers), [Context Switch](https://developer.arm.com/documentation/den0024/a/The-Memory-Management-Unit/Context-switching).

Reply:
Actually FP LR SP are included in the struct, only the LR is actually save in the field of name `pc`

Reply:
wouldn't you want to save all the context x0-x31, especially when you do a conditional yield.
e.g.
```
 while( !check_asynchronous_event() ) {
     yield();
}
```

Reply:
I think not all the registers saved in X86 context too, I am not sure how the `yield` is supported, maybe it is supported by some signal mechanism.